tasks/docs.ts

```diff
@@ -1,37 +1,56 @@
 import gulp from "gulp";
 import DocsEngine from "carbonldp-ts-docs-engine";
 import path from "path";
+import del = require('del');
 
 const projectRootDir:string = path.resolve( __dirname, "../" );
 const sourceDir:string = path.resolve( projectRootDir, "src/" );
 const outputDir:string = path.resolve( projectRootDir, "docs/" );
 
-export const docsBuildProd:gulp.TaskFunction = () => {
+let environment: "development" | "production";
 
-	let options:DocsEngine.Options = {
-		src: sourceDir,
-		out: outputDir,
-		mode: "production" as "production",
-		logLevel: "info" as "info",
+export const docsClean:gulp.TaskFunction = () =>
+	del( [ outputDir ] );
+docsClean.displayName = "docs:clean";
+
+const setENV:( env:"development" | "production" ) => gulp.TaskFunction = env => {
+	const fn:gulp.TaskFunction = () => {
+		environment = env;
+		return Promise.resolve();
 	};
 
-	return DocsEngine.generate(options);
+	fn.displayName = "docs:set-env";
+
+	return fn;
 };
-docsBuildProd.displayName = "docs:build|prod";
 
-export const docsBuildDev:gulp.TaskFunction = () => {
+export const generateDocumentation:gulp.TaskFunction = () => {
 
 	let options:DocsEngine.Options = {
 		src: sourceDir,
 		out: outputDir,
-		mode: "development" as "development",
-		logLevel: "info" as "info",
+		mode: environment,
+		logLevel: "info",
 	};
 
 	return DocsEngine.generate(options);
 };
+generateDocumentation.displayName = "docs:build|prod";
```

Comment:
Follow same suggestion as in: https://github.com/CarbonLDP/carbonldp-js-sdk/pull/394#discussion_r336674339